cmd/juju/machine/upgradeseries_test.go

```diff
@@ -41,7 +41,7 @@ func (s *UpgradeSeriesSuite) runUpgradeSeriesCommandWithConfirmation(c *gc.C, co
 	// mock remote API
 	mockController := gomock.NewController(c)
 	mockUpgradeSeriesAPI := mocks.NewMockUpgradeMachineSeriesAPI(mockController)
-	mockUpgradeSeriesAPI.EXPECT().UpgradeSeriesPrepare(gomock.Any()).AnyTimes()
+	mockUpgradeSeriesAPI.EXPECT().UpgradeSeriesPrepare(gomock.Any(), gomock.Any(), gomock.Any()).AnyTimes()
```

Comment:
I'm surprised we don't care what parameters are being passed. I would think the machine Number and series and whether or not we are forcing is quite relevant to what we want to test.

Reply:
I concur, @ExternalReality had differing thoughts, conversation in pr 8822.

Reply:
@hmlanigan, @jameinel  Due to the nature of these tests we don't need to spy on the `UpgradeSeriesPrepareMethod` method so we use the `mock` as a `stub`.

At the risk of sounding didactic I would add that many times `spying` & `mocking` are not needed when all that is really needed is a stub. Where spying and mocking are concerned with invocations, arguments, and return values; stubs only "stub out" an external dependency. So here the tests are saying "I don't care what your doing UpgradeSeriesPerpare, I already trust that you work since you are tested somewhere else. Now get out of my way!" 

Here is what Bob Marin's [Little Mocker](https://8thlight.com/blog/uncle-bob/2014/05/14/TheLittleMocker.html) has to say about `spys` vs `stubs`:

> That [spys] sounds like coupling.

>>    It is! You have to be careful. The more you spy, the tighter you couple your tests to the implementation of your system. And that leads to fragile tests.

So in general its not a good idea to spy if all you need is a stub. Concretely, these test only make assertions about the proper handling of command line arguments and options and command line behavior that is solely based on those CLI argument and options. And as such the external dependencies can be stubbed rather than spied or mocked.